### ISSUE TITLE
Yoda Condition not being detected when setting property in object

### ISSUE THREAD

**Issue by darthvader666uk** [WordPress/WordPress-Coding-Standards#1992]
Looks like Yoda Condition checking didn't flag when setting a property to an object which I expected.  At First I thought it was either because it was inside HTML or Echo.  This is an example of what I expected to flag:

`<input name="enable-revisions-retention" id="enable-revisions-retention" type="checkbox" value ="true" <?php echo ($options->get_option('revisions-retention-enabled') == 'true') ? 'checked="checked"' : ''; ?> />`

Yet it didn't.  If i remove the `->get_option('revisions-retention-enabled')` section, then it flags exactly as I expected, just like this:

`<input name="enable-revisions-retention" id="enable-revisions-retention" type="checkbox" value ="true" <?php echo ($options == 'true') ? 'checked="checked"' : ''; ?> />`

Does Yoda Condition checks get ignored in objects? I am using version 2.3.0 for this So maybe there's a fix for the future?

Any advice / help would be appreciated.

**Comment by dingo-d** [comment]
Hi, @darthvader666uk can you fill the bug issue template? This will help us with debugging the issue. What PHP, PHP_CS version are you using?

Also if you can put the example of your config that would also help.

**Comment by GaryJones** [comment]
Would you expect `$options->get_option('revisions-retention-enabled') = 'true'` (assignment to a method call) to be valid?

**Comment by jrfnl** [comment]
Notwithstanding the bug which needs to be looked into, @darthvader666uk are you aware of the WP [`checked()`](https://developer.wordpress.org/reference/functions/checked/) function ? The code currently used in the above example is a prime candidate to be replaced with a call to `checked()`.

**Comment by darthvader666uk** [comment]
hey @jrfnl I am not no.  I shall have a look at that and see how if we can plan to change that. 

I will stillw ait on the outcome of this bug, I got code ready to test to make sure it works etc.

**Comment by darthvader666uk** [comment]
Just checking in.

I wondered if there was a timeframe for this bug by chance?

**Comment by jrfnl** [comment]
@darthvader666uk Thanks for the reminder, just had a closer look at the code sample and this is actually not a bug.

> When doing logical comparisons involving variables, always put the variable on the right side and put constants, literals, or function calls on the left side. If neither side is a variable, the order is not important.

Ref: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#yoda-conditions

The condition in your code sample does not involve a variable, but a function call, so the order does not matter.

If the code would be a comparison with a property of the object, it would have gotten flagged:
```php
<input name="enable-revisions-retention" id="enable-revisions-retention" type="checkbox" value ="true" <?php echo ($options->revisions_retention_enabled == 'true') ? 'checked="checked"' : ''; ?> />
```

Suggest: close.
